packages/search/src/features/search/service.ts

```diff
@@ -26,29 +50,58 @@ export const searchComposition = async (params: ISearchQuery) => {
     status,
     type,
     applicationLocationId = EMPTY_STRING,
+    eventLocationId = EMPTY_STRING,
+    gender = EMPTY_STRING,
     name = EMPTY_STRING,
+    ageCheck = EMPTY_STRING,
+    nameCombinations = [],
     createdBy = EMPTY_STRING,
     from = 0,
     size = DEFAULT_SIZE,
     sort = SortOrder.ASC
   } = params
-
-  return client.search({
-    type: DEFAULT_SEARCH_TYPE,
-    from,
-    size,
-    body: {
-      query: queryBuilder(
-        query,
-        trackingId,
-        contactNumber,
-        registrationNumber,
-        name,
-        applicationLocationId,
-        createdBy,
-        { event, status, type }
-      ),
-      sort: [{ dateOfApplication: sort }]
+  if (ageCheck === 'TRUE' && nameCombinations.length > 0) {
```

Comment:
Just to be on safe side, we should do: `ageCheck.toUpperCase() === 'TRUE'`

Reply:
Agreed